tools/clang/lib/SPIRV/AstTypeProbe.cpp

```diff
@@ -569,11 +569,6 @@ bool canTreatAsSameScalarType(QualType type1, QualType type2) {
           type2->isFloatingType()) ||
          (type2->isSpecificBuiltinType(BuiltinType::LitFloat) &&
           type1->isFloatingType()) ||
-         // Treat 'literal int' and 'int'/'uint' as the same
```

Comment:
I'm curious why this was deleted - from what I understand this affects more than just this case?

Reply:
Yes it does, that's why I had to patch the template code, but what I believe is worst case we generate more instructions, but at least they are correct.
But yes, it could break other paths.
I don't believe this is the best solution, but fixing the AST seems more complex/risky and impacts DXIL.
(https://github.com/microsoft/DirectXShaderCompiler/issues/4727#issuecomment-1373638890)

The internal test suite is still running on this one, but so far looks good.

Reply:
All tests are green on our end.